networkx/algorithms/time_dependent.py

```diff
@@ -0,0 +1,118 @@
+"""Time dependent algorithms."""
+
+from datetime import datetime
+
+import networkx as nx
+from networkx.utils import not_implemented_for
+
+__all__ = ["cd_index"]
+
+
+@not_implemented_for("undirected")
+@not_implemented_for("multigraph")
+def cd_index(G, node, time_delta=5, weight=None):
+    r"""Compute the CD index.
+
+    Calculates the CD index for the graph based on the given "focal patent" node
+    considering the patents after `time_delta` years.
```

Comment:
Can you use language that doesn't rely on the setting of patents?  Or is this cd_index only used for patents?